pkg/image/definition.go

```diff
@@ -60,6 +60,13 @@ type OperatingSystem struct {
 	Users      []OperatingSystemUser `yaml:"users"`
 	Systemd    Systemd               `yaml:"systemd"`
 	Suma       Suma                  `yaml:"suma"`
+	Packages   Packages              `yaml:"packages"`
+}
+
+type Packages struct {
+	PKGList  []string `yaml:"packageList"`
+	AddRepos []string `yaml:"additionalRepos"`
+	RegCode  string   `yaml:"registrationCode"`
```

Comment:
I'm not sure if this is the best structure to have. The more I look at it, the more I find the repetition (`packages: -> packageList`) weird.

The only alternative I can think of is something like this:
```
  rpm:
    packages:
    - package1
    - package2
    additionalRepositories:
    - repo1
    - repo2
    registrationCode: INTERNAL-USE-ONLY-foo-bar
```

Perhaps I'm nitpicking though so I'd love if @rdoxenham and @jdob also look at this.

Reply:
Yeah, I see what you're saying @atanasdinov, but I still like the `operatingSystem.packages` bit. I agree that `packages --> packageList` seems a little strange/repetitive, but I'd also like to avoid the use of `rpms` as we have a dedicated `rpms/` directory and that may also cause confusion.

Reply:
I also like the `packages` bit, mainly for the reasons that @rdoxenham described. However, in order to avoid the `packages --> packageList` confusion, we could perhaps do something like this:

```yaml
packages:
  install:
  - package1
  - package2
  additionalRepositories:
  - repo1
  - repo2
  registrationCode: INTERNAL-USE-ONLY-foo-bar
```
@atanasdinov WDYT?

Reply:
@rdoxenham You're right that `rpm` potentially collides with the `rpm` directory, however, we also have to keep in mind that we still want to side load RPMs, right? So it is more confusing (to me at least) if you have different names for the different options of the same functionality.

@ipetrov117 The issue with `install` is that it is a verb which doesn't quite sit right with `additionalRepositories` and `registrationCode` on the side... I don't have any better ideas though so I will approve to unblock.

Reply:
Yeah... verb aside, I think the flow does still make sense...

1. This is the definition of the desired state of the operating system packages
2. These are the packages I want to list to install
3. Here's where you can potentially pull these packages from, or help solve for dependencies
4. Here's my registration code so you can pull from SLE Micro & PackageHub repo's

We should likely let this one go :)